Browser agnostic Baseline site & content (#1706)
web-platform-dx/web-features: gh-pages/web-features.md

- Creating feature definitions, which identify and describe capabilities of the web.
- Generating Baseline support data, which summarizes the availability of web features across key browsers and releases.
- Publishing the [web-features npm package](https://www.npmjs.com/package/web-features), which bundles feature identifiers with Baseline statuses.

## What is a feature?

Think of features as individual components of the web platform that web developers can use to achieve specific tasks.

As such, a feature has no specific size. A feature might cover a single CSS property, an entire JavaScript APIs with multiple interfaces, methods, and properties, or a combination of CSS properties, HTML elements, and JavaScript APIs. As long as a web developer would use the feature to achieve a specific goal, it's a feature.

For example, the `fetch()` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.

## How does web-features relate to Baseline?

> The web-features project computes, reviews, and publishes Baseline statues. Therefore, web-features is the source of truth for Baseline statuses.

Reviewer: ```suggestion
The web-features project computes, reviews, and publishes Baseline statuses. Therefore, web-features is the source of truth for Baseline statuses.
```

---
permalink: web-features
---

> # Web features

Reviewer 2: ```suggestion
# web-features
```

- Creating feature definitions, which identify and describe capabilities of the web.
- Generating Baseline support data, which summarizes the availability of web features across key browsers and releases.
- Publishing the [web-features npm package](https://www.npmjs.com/package/web-features), which bundles feature identifiers with Baseline statuses.

## What is a feature?

Think of features as individual components of the web platform that web developers can use to achieve specific tasks.

As such, a feature has no specific size. A feature might cover a single CSS property, an entire JavaScript APIs with multiple interfaces, methods, and properties, or a combination of CSS properties, HTML elements, and JavaScript APIs. As long as a web developer would use the feature to achieve a specific goal, it's a feature.

For example, the `fetch()` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.

> ## How is a feature described?

Reviewer: I would actually remove this section. As a casual reader, I'm not interested in these internal details. I'm also not interested in this as a consumer of the package. If I'm looking to contribute, then yes, but in that case I would much rather go and read the contributor docs on the repo directly.

Author: removed it

Features in the [web-features repository](https://github.com/web-platform-dx/web-features/) are authored as YAML files that are stored in the [features](https://github.com/web-platform-dx/web-features/blob/main/features) directory. Each file corresponds to a single feature, and contains metadata about the feature, such as its name, description, specification, or support data.

For example, the `fetch()` API feature is described in the [features/fetch.yml](https://github.com/web-platform-dx/web-features/blob/main/features/fetch.yml) file.

> In addition to meta information about the feature, the various elements that come together to make the feature are identified using a list of [browser-compat-data](https://github.com/mdn/browser-compat-data) entries, from the MDN repository. This repository already holds fine-grained support data for most web platform features, this is used to build up each feature grouping, by allowing us to identify all of the component elements.

Reviewer 2: Some terminology adjustments and avoiding suggesting that we use the BCD repo directly, as opposed to the published package. Avoiding the ambiguous first person plural.

```suggestion
In addition to meta information about the feature, the various parts that come together to make the feature are identified using [@mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) (BCD) entries. BCD holds fine-grained support data for most web platform features; this is used to build up each feature definition, identifying all of the component parts.
```

See [What makes a feature](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md#what-makes-a-feature) for more details.

> ## How are features used?
> 
> The web features information is the basis that allows us to identify Baseline features, therefore anywhere that displays Baseline status should use this repository as the source of truth.

Reviewer 2: There are other applications for feature definitions. I think narrowing the scope of the question makes this easier to write and edit:

```suggestion
## How does web-features relate to Baseline?

The web-features project computes, reviews, and publishes Baseline statues. Therefore, web-features is the source of truth for Baseline statuses.
```

## Get involved

> It’s a huge effort to group the entire web platform in this way, and will be an ongoing one given that new web platform features are released every month. The community is encouraged to help with the grouping work. See the [contributing information](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md) for more details.

Reviewer 2: Avoiding the "group" terminology since we have groups and features and it's an occasional point of confusion. I'm also suggesting an explicit call to action.

```suggestion
It’s a huge effort to describe the entire web platform in this way and will be an ongoing one given that new web platform features are released every month. Help out by contributing feature definitions. See the [contributing information](https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md) for more details.
```
